### PR TITLE
Support OIDC for Azure

### PR DESCRIPTION
**Pull request by sat0ken**
Support OIDC login to Azure, same as AWS and GCP.

Fixes https://github.com/diggerhq/digger/issues/569

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -36,6 +36,19 @@ inputs:
...
       description: Service account to be used when the 'google-workload-identity-provider' input is specified)
       required: false
+  setup-azure:
+    description: Setup Azure
+    required: false
+    default: 'false'
+  azure-client-id:
+    description: Azure Client ID to be used for Azure OIDC auth
+    required: false
+  azure-tenat-id:
```
Comment (Spartakovic) [inline action.yml]: typo here

File: action.yml [hunk 2]
```diff
@@ -173,6 +186,14 @@ runs:
         aws-region: ${{ inputs.aws-region }}
       if: ${{ inputs.setup-aws == 'true' && inputs.aws-role-to-assume != '' }}
 
+    - name: Configure OIDC Azure credentials
+      uses: azure/login@v1
+      with:
+          client-id: ${{ inputs.azure-client-id }}
+          tenant-id: ${{ inputs.azure-tenat-id }}
```
Comment (Spartakovic) [inline action.yml]: and here

File: action.yml [hunk 3]
```diff
@@ -36,6 +36,19 @@ inputs:
...
+    required: false
+    default: 'false'
+  azure-client-id:
+    description: Azure Client ID to be used for Azure OIDC auth
+    required: false
+  azure-tenat-id:
+    description: AzureAD ID of the organization you are using
+    required: false
+  azure-subscription-id:
+    description: Subscrition ID of you are using
```
Comment (Spartakovic) [inline action.yml]: also typo here

Comment (sat0ken) [reply action.yml]: Thanks review and sorry for my typo miss.
Fix this commit.

https://github.com/diggerhq/digger/pull/572/commits/3dd709d8d65a067e042e79d717312bcd780d483f

**Review by motatoes — APPROVED** [review]

**Comment by motatoes** [comment]
@sat0ken Thank you for the contribution! this will be merged into @develop and can be tested there while we perform our additional tests, it should be released in version @v0.1.33 :)

**Pull request merged**